core/src/consensus.rs

```diff
@@ -829,12 +844,34 @@ mod test {
         };
         local.process_vote_unchecked(&vote);
         assert_eq!(local.votes.len(), 1);
-        let vote = Tower::new_vote(&local, 3, Hash::default(), Some(0));
+        let vote = Tower::new_vote(8, &local, 3, Hash::default(), Some(0)).unwrap();
         //first vote expired, so index should be 0
         assert_eq!(vote.0.slots, vec![3]);
         assert_eq!(vote.1, 0);
     }
 
+    #[test]
+    fn test_new_vote_threshold() {
+        let mut local = VoteState::default();
+        //network is censoring
+        let vote = Vote {
+            slots: vec![0, 1, 2, 3],
+            hash: Hash::default(),
+        };
+        local.process_vote_unchecked(&vote);
+        //vote 4 should be at threshold 5
+        assert_eq!(Tower::new_vote(4, &local, 4, Hash::default(), None), None);
```

Comment:
Maybe one more test case where `last_bank_slot` is not None to exercise that code path?